packages/@lwc/synthetic-shadow/src/polyfills/event/polyfill.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperties, isNull, isUndefined } from '@lwc/shared';
+import { pathComposer } from '../../3rdparty/polymer/path-composer';
+import { retarget } from '../../3rdparty/polymer/retarget';
+import { eventTargetGetter, eventCurrentTargetGetter } from '../../env/dom';
+import { eventsDispatchedDirectlyOnShadowRoot } from '../../faux-shadow/events';
+import { getShadowRoot, isHostElement } from '../../faux-shadow/shadow-root';
+import { EventListenerContext, eventToContextMap } from '../../faux-shadow/events';
+import { getNodeOwnerKey } from '../../shared/node-ownership';
+import { getOwnerDocument } from '../../shared/utils';
+
+function patchedTargetGetter(this: Event): EventTarget | null {
+    const originalTarget = eventTargetGetter.call(this);
+    if (!(originalTarget instanceof Node)) {
```

Comment:
@caridy would a brand check using `instanceof Node` would pose an issue with Locker VNext creating `Node` from a different document? Or would this be transparent from the LWC engine standpoint?